include/up-cpp/communication/NotificationSink.h

```diff
@@ -0,0 +1,80 @@
+// SPDX-FileCopyrightText: 2024 Contributors to the Eclipse Foundation
+//
+// See the NOTICE file(s) distributed with this work for additional
+// information regarding copyright ownership.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Apache License Version 2.0 which is available at
+// https://www.apache.org/licenses/LICENSE-2.0
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#ifndef UP_CPP_COMMUNICATION_NOTIFICATIONSINK_H
+#define UP_CPP_COMMUNICATION_NOTIFICATIONSINK_H
+
+#include <up-cpp/transport/UTransport.h>
+#include <up-cpp/utils/Expected.h>
+#include <uprotocol/v1/umessage.pb.h>
+#include <uprotocol/v1/uri.pb.h>
+#include <uprotocol/v1/ustatus.pb.h>
+
+#include <memory>
+#include <optional>
+#include <utility>
+
+namespace uprotocol::communication {
+
+/// @brief Interface for uEntities to receive notifications.
+///
+/// Like all L2 client APIs, the NotificationSink is a wrapper on top of the L1
+/// UTransport API; in this instance, it provides for the notification
+/// receiving half of the notification model.
+struct NotificationSink {
+	using ListenCallback = transport::UTransport::ListenCallback;
+
+	using StatusOrSink =
```

Comment:
```suggestion
	using SinkOrStatus =
```

?

Matches the ordering of the types the `Expected` is templated over.